packages/block-library/src/footnotes/format.js

```diff
@@ -104,12 +147,11 @@ export const format = {
 				// When there is no footnotes block in the post, create one and
 				// insert it at the bottom.
 				if ( ! fnBlock ) {
-					const clientId = getSelectedBlockClientId();
-					let rootClientId = getBlockRootClientId( clientId );
+					let rootClientId = getBlockRootClientId( selectedClientId );
 
 					while (
 						rootClientId &&
-						getBlockName( rootClientId ) !== 'core/post-content'
+						getBlockName( rootClientId ) !== POST_CONTENT_BLOCK_NAME
```

Comment:
Nit: Why do we still need a while loop here if we got the parent post content block earlier?

Reply:
In other words, if we know if there's a post content block, we can append it there, if there's no such block, append it to `''` (the root)?

Reply:
Oh, good point. 

I'm testing replacing with 

```js
					if ( parentPostContent.length ) {
						rootClientId = parentPostContent[ 0 ];
					}
```

Working okay so far 👍🏻

Reply:
I had to revert this change because adding footnotes to list blocks was failing. I suspect this test doesn't account for nested blocks. Given we're close to 6.3 release,  we can do a follow up to see if we can tidy this.

Reply:
The failing test that brought this to our attention was `Footnotes › can be inserted in a list` for the record. Great to have tests!!